packages/slate/src/interfaces/element.ts

```diff
@@ -58,6 +66,20 @@ export const Element: ElementInterface = {
     return (props as Partial<Element>).children !== undefined
   },
 
+  /**
+   * Check if a value implements the `Element` interface and has type key
+   */
+
+  isElementType: (
+    value: any,
+    type: string
+  ): value is Element & { type: string } => {
+    if (value?.type === type && isElement(value)) {
+      return true
+    }
+    return false
+  },
```

Comment:
Maybe do something like the below to make the func more flexible?

```suggestion
  isElementType: (
    value: any,
    discriminantVal: string,
    discriminantKey: string = 'type'
  ): value is Element & { type: string } => {
    return isElement(value) && value[discriminantKey] === discriminantVal;
  },
```

Reply:
The point being that not everyone uses `type` (although most people do) so maybe we make the func flexible to other [discriminants](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions).

Reply:
haaa i see, now im understand why `type` is not in Element type